qutebrowser/keyinput/modeparsers.py

```diff
@@ -169,11 +172,56 @@ def __init__(self, win_id, mode, parent=None):
         """
         super().__init__(win_id, parent)
         self._read_config(mode)
+        self._orig_sequence = keyutils.KeySequence()
         self._mode = mode
+        self._ignore_next_key = False
 
     def __repr__(self):
         return utils.get_repr(self, mode=self._mode)
 
+    def handle(self, e, *, dry_run=False):
+        """Override to pass the chain through on NoMatch
+
+        Args:
+            e: the KeyPressEvent from Qt.
+            dry_run: Don't actually execute anything, only check whether there
+                     would be a match.
+
+        Return:
+            A self.Match member.
+        """
+        if keyutils.is_modifier_key(e.key()) or self._ignore_next_key:
+            self._ignore_next_key = self._ignore_next_key and dry_run
+            return QKeySequence.NoMatch
+
+        orig_sequence = self._orig_sequence.append_event(e)
+        match = super().handle(e, dry_run=dry_run)
+
+        if not dry_run and match == QKeySequence.PartialMatch:
+            self._orig_sequence = orig_sequence
+
+        if dry_run or len(orig_sequence) == 1 or match != QKeySequence.NoMatch:
+            return match
+
+        window = QApplication.focusWindow()
+        if window is None:
+            return match
+
+        self._ignore_next_key = True
```

Comment:
This seems rather brittle, as you're not guaranteed that the event you posted will be the next event which is received. I haven't tested it, but it likely also breaks if there's more than one key in the original sequence.

I'm okay with it when there's no other solution, but I'd like to check whether there's a better way. Can you try adding an attribute to the `press_event` / `release_event` object and see whether it's still there (via `getattr(e, 'ignore', False)` or so) when you get it back?